src/include/processor/operator/var_length_extend/bfs_state.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "processor/operator/result_collector.h"
+
+namespace kuzu {
+namespace processor {
+
+enum VisitedState : uint8_t {
+    NOT_VISITED = 0,
+    VISITED = 1,
+};
+
+struct BFSLevel {
+    std::vector<common::offset_t> nodeOffsets;
+
+    inline void resetState() { nodeOffsets.clear(); }
+    inline uint64_t size() const { return nodeOffsets.size(); }
+};
+
+struct SSPMorsel {
+    // Level state
+    uint8_t currentLevelNumber;
+    uint64_t currentLevelNodeIdx; // controls scan of current BFS level
+    std::unique_ptr<BFSLevel> currentBFSLevel;
+    std::unique_ptr<BFSLevel> nextBFSLevel;
+
+    // Visited state
+    uint64_t numVisitedNodes;
+    uint8_t* visitedNodes;
+    std::unique_ptr<uint8_t[]> visitedNodesBuffer;
+    uint8_t* distance;
+    std::unique_ptr<uint8_t[]> distanceBuffer;
+
+    // Start offset of src node. We need to exclude start node from result.
```

Comment:
Can you say some more? Why do we need to exclude start from result? Is that a semantics specific to shortest paths? Let's explain because we will forget.